src/ctl.c

```diff
@@ -3348,3 +3370,56 @@ experimental_utilization_batch_query_ctl(tsd_t *tsd, const size_t *mib,
 label_return:
 	return ret;
 }
+
+static const ctl_named_node_t *
+experimental_arenas_i_index(tsdn_t *tsdn, const size_t *mib,
+    size_t miblen, size_t i) {
+	const ctl_named_node_t *ret;
+
+	malloc_mutex_lock(tsdn, &ctl_mtx);
+	if (ctl_arenas_i_verify(i)) {
+		ret = NULL;
+		goto label_return;
+	}
+	ret = super_experimental_arenas_i_node;
+label_return:
+	malloc_mutex_unlock(tsdn, &ctl_mtx);
+	return ret;
+}
+
+static int
+experimental_arenas_i_pactivep_ctl(tsd_t *tsd, const size_t *mib,
+    size_t miblen, void *oldp, size_t *oldlenp, void *newp, size_t newlen) {
+	if (!config_stats) {
+		return ENOENT;
+	}
+	if (*oldlenp != sizeof(size_t *)) {
```

Comment:
It might be better to have `oldlenp == NULL || *oldlenp != sizeof(size_t *)`, just in case `oldlenp` happens to be `NULL`?

`oldp == NULL` can also be tested here to be rigorous, but the `READ(pactivep, size_t *)` below tests it anyway, so it doesn't make much difference, only except that it may save a few operations to test it here if the input is invalid.

Reply:
Good point. Will update.